mcodingbot/plugins/python.py

```diff
@@ -7,7 +7,7 @@
 
 plugin = Plugin()
 
-PEP_REGEX = re.compile(r"pep *(?P<pep>\d{1,4})", re.IGNORECASE)
+PEP_REGEX = re.compile(r"pep[\s-]*(?P<pep>\d+)", re.IGNORECASE)
```

Comment:
```suggestion
PEP_REGEX = re.compile(r"pep[\s-]*(?P<pep>\d{1,4}[^\d])", re.IGNORECASE)
```

Reply:
Throwing a word boundary at the end also works if you dont want to match `pep 1234abc`.

Reply:
trag1c's is better because `pep 9.` should match, but you don't want `pep 99999` to become `pep 9999`

Reply:
This one won't work, requires another character after the PEP number

Reply:
<img width="554" alt="image" src="https://user-images.githubusercontent.com/77130613/189768773-51490979-d525-401a-955e-86793d99f4e5.png">

Reply:
![image](https://user-images.githubusercontent.com/65521138/189769067-a1b4dc2a-2653-4e36-b0e6-7566c2ed43f5.png)


Word boundary works

Reply:
> trag1c's is better because `pep 9.` should match, but you don't want `pep 99999` to become `pep 9999`

pep 99999 doesn't match to begin with for mine

Reply:
How does `?P` even work? What is it for?

Reply:
> How does `?P` even work? What is it for?

Named capture group; in other syntaxes it's just `?` I believe

Reply:
> > trag1c's is better because `pep 9.` should match, but you don't want `pep 99999` to become `pep 9999`
> 
> pep 99999 doesn't match to begin with for mine

Yes but your method breaks if `pep <x>` isn't surrounded by spaces. We don't want that. trag1c's method works for every case so there's no reason to re-engineer the regex.

Reply:
> How does `?P` even work? What is it for?

https://stackoverflow.com/questions/10059673/named-regular-expression-group-pgroup-nameregexp-what-does-p-stand-for

Reply:
![image](https://user-images.githubusercontent.com/65521138/189770176-910b7b79-0b40-4936-a1ee-f31d4fa58660.png)
Mine fails in this case but tragic's does as well for some reason. Mine does work with trailing spaces.
anyway why does that fail lol

Reply:
```py
REGEX = re.compile(r"pep[\s-]*(?P<pep>\d{1,4}\b)", re.IGNORECASE)
```

Since Lunar's regex works for all the cases I can find, we wouldn't have to check that the regex is < 1000 inside comp.